src/redux/books/booksSlice.js

```diff
@@ -1,53 +1,69 @@
-import { createSlice, nanoid } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
 
 const initialState = {
-  books: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  loading: false,
+  error: '',
 };
-
+const appId = 'xEU2P3EW7ZDOC4O4U6Sh';
+const url = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${appId}`;
+export const addBookToServer = createAsyncThunk('books/addToServer', async (obj) => {
+  try {
+    const response = await axios.post(`${url}/books`, obj);
+    const { data } = response;
+    return data;
+  } catch (error) {
+    return error;
+  }
+});
+export const getBookFromServer = createAsyncThunk('books/getFromServer', async () => {
+  try {
+    const response = await axios.get(`${url}/books`);
+    const { data } = response;
+    return data;
+  } catch (error) {
+    return error;
+  }
+});
+export const deleteBookFromServer = createAsyncThunk('books/deleteFromServer', async (obj) => {
+  const { item_id: id } = obj;
+  try {
+    const response = await axios.delete(`${url}/books/${id}`, {
+      data: obj,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    const { data } = response;
+    return data;
+  } catch (error) {
+    return error;
+  }
+});
```

Comment:
- [ ] It would be nice if your app should not reload to render the latest data because whenever I add or remove something then I need to reload the page to show the latest data 😨 Kindly make this change to make your app behave like a single page app which works as expected in the project requirements 👍